doc/release-notes/0.40/0.40.md

```diff
@@ -0,0 +1,122 @@
+
+<table cellpadding="4" cellspacing="0" summary="" width="100%" rules="all" frame="border" border="1"><thead align="left">
+<tr>
+<th valign="bottom">Issue number</th>
+<th valign="bottom">Description</th>
+<th valign="bottom">Version / Platform</th>
+<th valign="bottom">Impact</th>
+</tr>
+</thead>
+<tbody>
+
+<tr>
+<td valign="top"><a href="https://github.com/eclipse-openj9/openj9/issues/17073">#17073</a></td>
+<td valign="top">Support for persistent shared classes cache is now added on z/OS&reg; systems</td>
+<td valign="top">All versions (z/OS)</td>
+<td valign="top">Persistent shared caches are now supported on z/OS systems. With the 31-bit VM on all versions of z/OS, the shared cache is memory mapped within the 0-2 GB address range. With the 64-bit VM on z/OS version 2.3 and earlier, the persistent shared cache is memory mapped within the 0-2 GB address range but on z/OS version 2.4 and later the shared cache is memory mapped above the 2 GB address range. </td>
+</tr>
+
+<tr>
+<td valign="top"><a href="https://github.com/eclipse-openj9/openj9/issues/17425">#17425</a></td>
+<td valign="top">A new <tt>-XX:codecachetotalMaxRAMPercentage</tt> option is added to set the maximum limit for the total JIT code cache size based on the amount of physical memory available to the VM.</td>
+<td valign="top">All versions</td>
+<td valign="top">The new option is used to set the JIT code cache maximum size limit as a percentage of the available physical memory. In memory constrained environments, the percentage limit prevents the VM from using too much memory for its JIT code caches, which was thus, leaving too little memory for other VM or JIT data structures. By default, the VM is not allowed to use more than 25% of the available physical memory for its JIT code caches.</td>
+</tr>
+
+<tr>
+<td valign="top"><a href="https://github.com/eclipse/omr/issues/7021">#7021</a></td>
+<td valign="top">If you mount the <tt>/proc</tt> file system with the <tt>hidepid=2</tt> option on Linux&reg; systems and the VM does not have root privileges, it cannot access the /proc file system, and the container detection fails. But, the VM does not exit as it did in the previous releases. Now, the VM starts with a warning message.</td>
+<td valign="top">All versions (Linux)</td>
+<td valign="top">Although the VM starts after the container detection fails, the VM assumes that it is running outside a container. Therefore, if the VM is running in a container, the VM cannot adapt to the container's limitations and might use an undesirable amount of resources. You can evaluate the impact on performance because of the container detection failure and take steps to resolve the performance issue, if so required. For example, you can remount the <tt>/proc</tt> file system with the <tt>gid</tt> and <tt>hidepid=2</tt> options to allow some processes to access the <tt>/proc</tt> file system. </td>
```

Comment:
minor nitpick:
```suggestion
<td valign="top">Although the VM starts after the container detection fails, the VM assumes that it is running outside a container. Therefore, if the VM is running in a container, the VM cannot adapt to the container's limitations and might use an undesirable amount of resources. You can evaluate the impact on performance because of the container detection failure and take steps to resolve the performance issue, if so required. For example, you can remount the <tt>/proc</tt> file system with the <tt>gid</tt> and <tt>hidepid=2</tt> options to allow some processes to access the <tt>/proc</tt> file system.</td>
```